### DISCUSSION TITLE
Clarification in `delete-user.js`

### DISCUSSION THREAD

**Discussion by MaheshkumarSundaram** [neonexus/sails-react-bootstrap-webpack#137] [Q&A] [answered]
Hi @neonexus 

In `delete-user.js` controller,

```js
await sails.models.user.update({id: foundUser.id}).set(_.merge({}, foundUser, {
            deletedAt: moment.tz(sails.config.datastores.default.timezone).toDate(),
            deletedBy: env.req.session.user.id
        }));
```

Why are you merging the existing user data with `deletedAt` & `deletedBy`? Is it because to restrict the automatic update of `updatedAt`?

**Comment by neonexus** [comment, marked as answer]
@MaheshkumarSundaram sorry for delay; haven't been keeping up with open-source stuff as of late.

Frankly, looking at this, the merge is not needed; I don't know why this is here. However, `deletedAt` and `deletedBy` are special "soft-delete" attributes. `updatedAt` has no bearing here; it will always update on a record update.

**Reply by MaheshkumarSundaram** [reply]
Alright thanks for the clarification
